### PR TITLE
Transcribing backend flag

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: src/main/java/org/jitsi/jigasi/JvbConference.java [hunk 1]
```diff
@@ -729,6 +724,50 @@ else if (evt.getNewState() == RegistrationState.CONNECTION_FAILED)
...
+                        .discoverInfo(JidCreate.domainBareFrom(this.callContext.getRoomJidDomain()));
+
+                DiscoverInfo.Identity avIdentity = info.getIdentities().stream().
+                    filter(di -> di.getCategory().equals("component") && di.getType().equals("av_moderation"))
+                        .findFirst().orElse(null);
+
+                if (avIdentity != null && this.getAudioModeration() != null)
+                {
+                    String avModerationAddress = avIdentity.getName();
+                    this.getAudioModeration().setAvModerationAddress(avModerationAddress);
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/JvbConference.java:764]: I think you need a null check, `this.audioModeration` is set to null in some cases

File: src/main/java/org/jitsi/jigasi/JvbConference.java [hunk 2]
```diff
@@ -729,6 +724,50 @@ else if (evt.getNewState() == RegistrationState.CONNECTION_FAILED)
...
+
+                    logger.info(String.format("%s Discovered %s for %oms.",
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/JvbConference.java:766]: Do we really care about this message only for `avModarationAddress`? I suggest either remove it or move it out of the "if" to be printed right after the `discoverInfo` call

File: src/main/java/org/jitsi/jigasi/JvbConference.java [hunk 3]
```diff
@@ -760,6 +766,24 @@ private void discoverComponentAddresses()
...
                 }
+
+                DiscoverInfo.Identity roomMetadataIdentity = info.getIdentities().stream().
+                        filter(di -> di.getCategory().equals("component") && di.getType().equals("room_metadata"))
+                        .findFirst().orElse(null);
+
+                // we process room metadata messages only when we are transcribing
+                if (roomMetadataIdentity != null && this.gatewaySession instanceof TranscriptionGatewaySession)
+                {
+                    if (roomMetadataListener != null)
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/JvbConference.java:777]: Any reason not to make the field `final` and always initialize it? You can remove these 5 lines of code and not worry about this executing after a re-connect for example

File: src/main/java/org/jitsi/jigasi/TranscriptionGatewaySession.java [hunk 4]
```diff
@@ -308,8 +320,14 @@ void notifyChatRoomMemberUpdated(ChatRoomMember chatMember, Presence presence)
...
 
-        if (transcriber.isTranscribing() &&
-            !transcriber.isAnyParticipantRequestingTranscription())
+        this.maybeStopTranscription();
+    }
+
+    private void maybeStopTranscription()
+    {
+        if (transcriber.isTranscribing()
+                && !(transcriber.isAnyParticipantRequestingTranscription()
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/TranscriptionGatewaySession.java:329]: nit: suggest to extract a `isTranscriptionRequested()` method

File: src/main/java/org/jitsi/jigasi/JvbConference.java (hunk 1 above)
Comment (damencho) [reply src/main/java/org/jitsi/jigasi/JvbConference.java:764]: But there is a check in the if?

File: src/main/java/org/jitsi/jigasi/JvbConference.java (hunk 2 above)
Comment (damencho) [reply src/main/java/org/jitsi/jigasi/JvbConference.java:766]: I'll remove that and move the print after, I just kept it when moving the stuff.

File: src/main/java/org/jitsi/jigasi/JvbConference.java (hunk 1 above)
Comment (bgrozev) [reply src/main/java/org/jitsi/jigasi/JvbConference.java:766]: Ops, I missed that

**Review by bgrozev — APPROVED** [review]

**Pull request merged**